### PR TITLE
Nested Record Namespace Changes

### PR DESCRIPTION
**Pull request by ethan-schwab**
For records with nested fields, avro-to-pyton library is currently creating a version of the nested record in every directory it is referred to, as opposed to one time in its own directory (namespace). The types/record.py class currently uses the parent_namespace field like all the other types (map, array, primitive etc) but I suggest we change it to use the namespace of the record itself. These changes also address a similar issue with enums and union fields

Example:
**Current output:**
Record X {
 namespace x
 NestedRecordY {
   namespace x
 }
}

**Desired Output:**
Record X {
 namespace x
 NestedRecordY {
  namespace y
 }
}

### CODE REVIEW

File: avro_to_python/cli.py
Comment (ethan-schwab) [inline avro_to_python/cli.py:6]: this was a hack to run the job locally ill remove it before merging

Comment (chasdevs) [inline avro_to_python/cli.py]: Assuming this would just be a temp thing (along with the other references to your local env)

File: tests/writer/test_duplicate_namespace.py
Comment (chasdevs) [inline tests/writer/test_duplicate_namespace.py:42]: Looks good; we'd also want to test that the bundled schema itself has the correct `namespace` values

**Comment by ethan-schwab** [comment]
hi @SRserves85 when you get a chance can you please take a look at this? thanks!

**Review by SRserves85 — APPROVED** [review]

**Comment by ethan-schwab** [comment]
hi Scott, had to add a one line change to the unit test to make it work whether or not the output files exist already. Your approval still stands but I need a maintainer to approve the running workflow so I can merge. Sorry about that thanks! 

@SRserves85

**Review by chasdevs — APPROVED** [review]

**Pull request merged**
